frontend/pages/_app.tsx

```diff
@@ -15,13 +16,24 @@ type AppPropsWithLayout<P> = AppProps<P> & {
    Component: NextPageWithLayout;
 };
 
-if (process.env.NEXT_PUBLIC_MOCK_API === 'true') {
-   import('@tests/playwright/msw').then(({ setupMocks }) => {
-      setupMocks();
-   });
-}
-
 function MyApp({ Component, pageProps }: AppPropsWithLayout<any>) {
+   const [mswState, setMswState] = useState<'unused' | 'loading' | 'ready'>(
+      'unused'
+   );
+
+   if (process.env.NEXT_PUBLIC_MOCK_API === 'true' && mswState === 'unused') {
```

Comment:
Not extremely familiar with this, but didn't you set the state right above? Won't `mswState === 'unused'` always be true here?